framework/kms-keyring.md

```diff
@@ -207,11 +209,11 @@ as described above, OnEncrypt MUST also attempt to encrypt the plaintext data ke
 
 To attempt to encrypt the plaintext data key with a particular CMK, OnEncrypt MUST call [KMS Encrypt](#kms-encrypt).
 
-The [KMS client](#kms-client) that calls [KMS Encrypt](#kms-encrypt) MUST be the
-client returned by the [client supplier](#client-supplier).
-The client MUST be a client that calls the AWS region specified in the [generator](#generator) ARN.
-If the [client supplier](#client-supplier) does not provide any client for the given region for this Encrypt call,
-OnEncrypt MUST skip that particular CMK.
+For each [KMS Encrypt](#kms-encrypt) call, if an AWS region can be extracted from the [Key Id](#key-ids), then the
+[KMS client](#kms-client) that calls [KMS Encrypt](#kms-encrypt) MUST be the client returned by the
+[client supplier](#client-supplier) when given that region as input.
+If an AWS region cannot be extracted from the CMK ARN then the KMS Keyring MUST input a value denoting an unknown region.
```

Comment:
```suggestion
If an AWS region cannot be extracted from the Key ID then the KMS Keyring MUST input a value denoting an unknown region.
```

Reply:
"CMK ARN" is more consistent with the rest of the document. "Key ID" elsewhere is only used to refer elements of the "Key IDs" list.

In my opinion we should revise the spec to have better wording around "CMK" vs "CMK ARN" vs "Key ID". But that is not within the scope of this PR.

Reply:
Actually, re-reading through the file, I see the "Key ID" is more consistent within the paragraph. I will add this change.